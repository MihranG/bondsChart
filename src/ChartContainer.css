.chart-wrapper{
    align-self: center;
    display: flex;
}