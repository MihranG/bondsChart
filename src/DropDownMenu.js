import React from "react";
import { connect } from "react-redux";

import { setActiveMode } from "./store";

const DropDownMenuDisconnected = ({ activeViewMode, handleChange }) => {
  return (
    <div className="dropdown-wrapper">
      <select onChange={handleChange} value={activeViewMode}>
        <option value="yield">Yield</option>
        <option value="spread">Spread</option>
        <option value="price">Price</option>
      </select>
    </div>
  );
};

const mapStateToProps = state => ({
  activeViewMode: state.viewMode.activeMode
});
const mapDispatchToProps = dispatch => ({
  handleChange: e => dispatch(setActiveMode(e.target.value))
});

const DropDownMenu = connect(
  mapStateToProps,
  mapDispatchToProps
)(DropDownMenuDisconnected);

export default DropDownMenu;
