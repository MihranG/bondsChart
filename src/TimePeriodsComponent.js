import React from "react";

import "./TimePeriods.css";
import { connect } from "react-redux";

import { setActiveTimePeriod } from "./store";
import { fetchData } from "./thunks";

const TIMELAPSES = [
  { id: "week", name: "Week" },
  { id: "month", name: "Month" },
  { id: "quarter", name: "Quarter" },
  { id: "year", name: "Year" },
  { id: "max", name: "Max" }
];

function TimeLapseComponentDisconnected({
  timePeriodHandler,
  fetchData,
  activePeriod
}) {
  const clickHandler = e => {
    timePeriodHandler(e.target.id);
    fetchData(e.target.id)();
  };
  return (
    <div className="buttons-wrapper">
      {TIMELAPSES.map(time => (
        <button
          className={
            activePeriod === time.id
              ? "active periods-button"
              : "periods-button"
          }
          key={time.id}
          id={time.id}
          onClick={clickHandler}
        >
          {time.name}
        </button>
      ))}
    </div>
  );
}

const mapStateToProps = state => ({
  activePeriod: state.time.activePeriod
});

const mapDispatchtoProps = dispatch => ({
  timePeriodHandler: id => dispatch(setActiveTimePeriod(id)),
  fetchData: period => dispatch(fetchData(period))
});

const TimePeriodsComponent = connect(
  mapStateToProps,
  mapDispatchtoProps
)(TimeLapseComponentDisconnected);

export default TimePeriodsComponent;
