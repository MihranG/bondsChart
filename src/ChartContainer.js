import React from "react";
import "./ChartContainer.css";
import {
  LineChart,
  YAxis,
  XAxis,
  Tooltip,
  CartesianGrid,
  Line
} from "recharts";
import { connect } from "react-redux";

import { fetchData } from "./thunks";

export class ChartContainerDisconnected extends React.Component {
  componentDidMount() {
    const { data, fetchData, activePeriod } = this.props;
    if (data.length === 0) {
      fetchData(activePeriod)();
    }
  }

  render() {
    const { data, isLoading, activeViewMode } = this.props;
    return (
      <div className="chart-wrapper">
        {isLoading ? (
          <p id="loading_text" test-id="loading">
            Loading...
          </p>
        ) : (
          <LineChart
            width={400}
            height={400}
            data={data}
            margin={{ top: 5, right: 20, left: 10, bottom: 5 }}
          >
            <XAxis dataKey="name" />
            <YAxis />
            <Tooltip />
            <CartesianGrid stroke="#f5f5f5" />
            <Line
              type="monotone"
              dataKey={activeViewMode}
              stroke="#ff7300"
              yAxisId={0}
            />
          </LineChart>
        )}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    data: state.bondsData.data,
    isLoading: state.time.isLoading,
    activePeriod: state.time.activePeriod,
    activeViewMode: state.viewMode.activeMode
  };
}
function mapDispatchToProps(dispatch) {
  return {
    fetchData: period => dispatch(fetchData(period))
  };
}

const ChartContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ChartContainerDisconnected);

export default ChartContainer;
