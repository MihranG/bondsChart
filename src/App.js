import React, {useState,useEffect} from 'react';
import './App.css';

import {priceApiGetterStandard} from './fakeApi/fakeApiGetter'

import ChartContainer from './ChartContainer'
import TimePeriodsComponent from './TimePeriodsComponent'

function App() {
  const [timePeriod, setTimePeriod] = useState('week');
  const [bondData, setBondData] = useState([])
  useEffect(()=>{
    priceApiGetterStandard(timePeriod).then(res=>{
      setBondData(res)
    }).catch(e=>console.error(e))
  },[timePeriod])

  return (
    <div className="App">
      <TimePeriodsComponent timePeriodHandler={setTimePeriod}  activePeriod={timePeriod}/>
      <ChartContainer data={bondData} />
    </div>
  );
}

export default App;
