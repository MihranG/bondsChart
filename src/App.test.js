import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import ChartContainer from "./ChartContainer";
import DropDownMenu from "./DropDownMenu";
import TimePeriodsComponent from "./TimePeriodsComponent";

describe("App component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />);
  });
  it("should match  snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  afterAll(() => {
    wrapper.unmount();
  });
});

describe("Whole application", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<App />);
  });
  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have appropriate components", () => {
    expect(wrapper.find(ChartContainer).get(0)).not.toBe(undefined);
    expect(wrapper.find(DropDownMenu).get(0)).not.toBe(undefined);
    expect(wrapper.find(TimePeriodsComponent).get(0)).not.toBe(undefined);
  });

  afterAll(() => {
    wrapper.detach();
  });
});
