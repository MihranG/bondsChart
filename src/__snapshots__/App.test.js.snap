// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should match  snapshot 1`] = `ShallowWrapper {}`;

exports[`Whole application should match snapshot 1`] = `ReactWrapper {}`;
